buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -46,6 +48,15 @@ public void apply(Project project) {
         File compilerJavaHome = findCompilerJavaHome();
         File runtimeJavaHome = findRuntimeJavaHome(compilerJavaHome);
 
+        String testSeedProperty = System.getProperty("tests.seed");
```

Comment:
I feel like this could be simplified to something like this which seems easier to grok:
```
String testSeed = System.getProperty("tests.seed", generateSeed());

private static String generateSeed() {
    double seed = new Random().nextLong();
    return Long.toUnsignedString(seed, 16).toUpperCase(Locale.ROOT);
}
```

Reply:
That would mean always generating seed, even when one is passed in as a property. If you are adamant about that, I'll let you make that change with the other refactorings to global build info we discussed.